src/sentry/static/sentry/app/components/modals/addDashboardWidgetModal.tsx

```diff
@@ -215,10 +215,28 @@ class AddDashboardWidgetModal extends React.Component<Props, State> {
       const newState = cloneDeep(prevState);
       newState.queries.splice(index, index + 1);
 
+      // If there is only one query, then its name will not be visible.
+      if (newState.queries.length === 1) {
+        newState.queries[0].name = '';
+      }
+
+      return newState;
+    });
+  };
+
+  handleAddOverlay = () => {
+    this.setState(prevState => {
+      const newState = cloneDeep(prevState);
+      newState.queries.push(cloneDeep(newQuery));
+
       return newState;
     });
   };
 
+  canAddOverlay() {
+    return ['line', 'area', 'stacked_area', 'bar'].includes(this.state.displayType);
```

Comment:
Should we prune off queries for tables/big number/worldmap when the chart type is changed?

Reply:
Queries are already pruned in `handleFieldChange()`.